modules/web/src/app/dynamic/enterprise/cluster-backups/list/restore/template.html

```diff
@@ -0,0 +1,208 @@
+        </th>
+        <td mat-cell
+            *matCellDef="let element">
+          <span>{{element.spec.backupName}}</span>
+        </td>
+      </ng-container>
+      <ng-container matColumnDef="namespaces">
+        <th mat-header-cell
+            *matHeaderCellDef
+            class="km-header-cell p-15">Namespaces
+        </th>
+        <td mat-cell
+            *matCellDef="let element">
+          <span>{{element.spec.includedNamespaces?.length ?? 'all'}}</span>
+        </td>
+      </ng-container>
+      <ng-container matColumnDef="created">
+        <th mat-header-cell
+            *matHeaderCellDef
+            class="km-header-cell p-15">Created
+        </th>
+        <td mat-cell
+            *matCellDef="let element">
+          <km-relative-time [date]="element.spec.createdAt"></km-relative-time>
+        </td>
+      </ng-container>
+      <ng-container matColumnDef="actions">
+        <th mat-header-cell
+            *matHeaderCellDef
+            class="km-header-cell"></th>
+        <td mat-cell
+            *matCellDef="let element">
+          <div fxLayoutAlign="end"
+               class="km-table-actions">
+            <button mat-icon-button
+                    matTooltip="Delete Restore"
+                    (click)="deleteRestores([element]); $event.stopPropagation()">
+              <i class="km-icon-mask km-icon-delete"></i>
+            </button>
+          </div>
+        </td>
+      </ng-container>
+      <ng-container matColumnDef="nameSpacesDetails">
+        <td mat-cell
+            *matCellDef="let element"
+            [attr.colspan]="columns.length">
+          <div class="namespaces-detail">Namespaces</div>
+          <div>
+            <mat-chip *ngFor="let namespace of element.spec.includedNamespaces"
+                      class="namespaces-label">
+              <div>{{namespace}}</div>
+            </mat-chip>
+          </div>
+        </td>
+      </ng-container>
+
+      <tr mat-header-row
+          class="km-header-row"
+          *matHeaderRowDef="columns"></tr>
+      <tr mat-row
+          *matRowDef="let element; columns: columns"
+          class="km-mat-row km-pointer"
+          [ngClass]="{'km-selected': checkSelected(element.id)}"
+          (click)="toggelRestoreDetail(element.name)"></tr>
+      <tr mat-header-row
+          *matRowDef="let element; columns: toggleableColumn"
+          [ngClass]="isRestoreToggeled(element.name) ? '' : 'km-hidden'"
+          class="km-mat-row"></tr>
+    </table>
+
+    <div class="km-row km-empty-list-msg"
+         *ngIf="!clusterRestores?.length && !loadingRestores">
+      No restores available.
+    </div>
+
+    <div class="km-row"
+         *ngIf="loadingRestores && !clusterRestores?.length">
+      <mat-spinner color="accent"
+                   class="km-spinner km-with-spacing"
+                   [diameter]="25"></mat-spinner>
+    </div>
+
+    <div [hidden]="!clusterRestores.length"
```

Comment:
Please add a check here or initialize `clusterRestores` with `[]` (similar to backups & schedules pages) because this throws an error when page is refreshed.

```suggestion
    <div [hidden]="!clusterRestores?.length"
```